pkg/k8s/k8s_namespace.go

```diff
@@ -0,0 +1,66 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// InitializeNamespace sets up the namespace based on the KNUU_DEDICATED_NAMESPACE environment variable
+func InitializeNamespace(identifier string) (string, error) {
+	useDedicatedNamespace, err := strconv.ParseBool(os.Getenv("KNUU_DEDICATED_NAMESPACE"))
+	if err != nil {
+		useDedicatedNamespace = false
+	}
+
+	var namespaceName string
+	if useDedicatedNamespace {
+		namespaceName = "knuu-" + sanitizeName(identifier)
+		logrus.Debugf("namespace random generated: %s", namespaceName)
+		if err := createNamespace(Clientset(), namespaceName); err != nil {
+			return "", fmt.Errorf("failed to create dedicated namespace: %v", err)
+		}
+
+		logrus.Debugf("full namespace name generated: %s", namespaceName)
+	} else {
+		namespaceName = "test"
+	}
+
+	return namespaceName, nil
+}
+
+// createNamespace creates a new namespace if it does not exist
+func createNamespace(clientset *kubernetes.Clientset, name string) error {
+	ctx := context.TODO()
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+
+	_, err := clientset.CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating namespace %s: %v", name, err)
+	}
+	return nil
+}
+
+// sanitizeName ensures that the namespace name complies with Kubernetes restrictions
+func sanitizeName(name string) string {
+	sanitized := strings.ToLower(name)
+	sanitized = strings.ReplaceAll(sanitized, "_", "-")
+
+	if len(sanitized) > 63 {
+		sanitized = sanitized[:63]
+	}
+
+	return sanitized
+}
```

Comment:
To cover all the edge cases as we use user input here:

```suggestion
// precompile the regular expression to avoid recompiling it on every function call
var invalidCharsRegexp = regexp.MustCompile(`[^a-z0-9-]+`)

// sanitizeName ensures compliance with Kubernetes DNS-1123 subdomain names. It:
// 1. Converts the input string to lowercase.
// 2. Replaces underscores and any non-DNS-1123 compliant characters with hyphens.
// 3. Trims leading and trailing hyphens.
// 4. Ensures the name does not exceed 63 characters, trimming excess characters if necessary
//    and ensuring it does not end with a hyphen after trimming.
// Use this function to sanitize strings to be used as Kubernetes names for resources.
func sanitizeName(name string) string {
	sanitized := strings.ToLower(name)
	// Replace underscores and any other disallowed characters with hyphens
	sanitized = invalidCharsRegexp.ReplaceAllString(sanitized, "-")
	// Trim leading and trailing hyphens
	sanitized = strings.Trim(sanitized, "-")

	if len(sanitized) > 63 {
		sanitized = sanitized[:63]
		// Ensure it does not end with a hyphen after cutting it to the max length
		sanitized = strings.TrimRight(sanitized, "-")
	}

	return sanitized
}
```

Reply:
Maybe move this to `k8s.go` as this could also be used from the other resources later.